pkg/bigquery/sanitize.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bigquery implements helpers for using BigQuery.
+package bigquery
+
+import (
+	"fmt"
+	"regexp"
+	"unicode/utf8"
+)
+
+// Start with lowercase, middle is lowercase, number or hyphen, cannot end in
+// hyphen. 6-30 characters in length (start, 4-28 middle, end).
+const projectIDRegex = `^[a-z][a-z0-9\-]{4,28}[a-z0-9]$`
+
+var projectIDMatcher = regexp.MustCompile(projectIDRegex)
```

Comment:
Optional: it might be cleaner to inline projectIDRegex into the MustCompile() call.

Reply:
I think I'll leave it as-is. The regex is already a long and hard to read line.